protos/flyteidl/admin/execution.proto

```diff
@@ -310,6 +310,9 @@ message ExecutionSpec {
     // If enabled, all calculations are performed even if cached results would be available, overwriting the stored
     // data once execution finishes successfully.
     bool overwrite_cache = 22;
+
+    // Environment variables to be set for the execution.
+    Envs envs = 23;
```

Comment:
What is the reason to use a separate struct for `Envs` rather than just setting this as:
```
map<string, string> environment_variables = 23;
```
It seems like just another layer of indirection?

Reply:
@EngHabu @hamersaw IIUC, the default value of map is {}. Adding another layer allows us to know value is set or unset. 
If people didn't add env to WorkflowExecutionConfig , the default value of env is always {} in it, and it may override the env in execution.

Reply:
In the case of `interruptible` support here it was beneficial to know whether the flag was set or unset. In this case, I don't think it is necessary or has any benefit. A not provided `env` struct will be treated the same as one with 0 length right?

Per @EngHabu comment below. It sounds like we should list this anyways - I'm OK with this route. So we will need the struct `EnvironmentVariable` and then update the references to `repeated EnvironmentVariable env_vars = x;` or something like that.

Reply:
we can't do something like below without adding another layer
```python
if override_env is not None:
    execution_env = override_env
```